nifi-extension-bundles/nifi-kafka-bundle/nifi-kafka-processors/src/main/java/org/apache/nifi/kafka/processors/PublishKafka.java

```diff
@@ -485,6 +485,9 @@ private void routeResults(final ProcessSession session, final List<FlowFileResul
 
             final Relationship relationship = flowFileResult.getExceptions().isEmpty() ? REL_SUCCESS : REL_FAILURE;
             session.transfer(flowFile, relationship);
+            final int numTopics = flowFileResult.getSentPerTopic().keySet().size();
+            final String topicList = String.join(",", flowFileResult.getSentPerTopic().keySet().toArray(new String[numTopics]));
```

Comment:
The `toArray` call is unnecessary because it'll take an `Iterable`. I'll fix and commit.